src/app/zap-templates/templates/app/cluster-objects-src.zapt

```diff
@@ -58,6 +59,29 @@ CHIP_ERROR DecodableType::Decode(TLV::TLVReader &reader) {
         {
             {{#zcl_command_arguments}}
             case to_underlying(Fields::k{{asUpperCamelCase label}}):
+                {{#if_is_enum type}}
+                {{#unless isArray}}
+                {{#if (isWeaklyTypedEnum (asUpperCamelCase type))}}
+// Need to convert consumers to using the new enum classes, so we
+// don't just have casts all over.
+#ifdef CHIP_USE_ENUM_CLASS_FOR_IM_ENUM
+                {{/if}}
+                {{#if isNullable}}
+                VerifyOrReturnError(!{{asLowerCamelCase label}}.IsNull(), CHIP_NO_ERROR);
```

Comment:
But ... this is not going to decode the actualy value, or indeed any values after this one.  Was this codepath tested?

Also, how does it make any sense to examine `{{asLowerCamelCase label}}` here _before_ we have done the decode into it?

Reply:
if it is null, there is no need for further decoding?

Reply:
yes, you are right, fixing it.